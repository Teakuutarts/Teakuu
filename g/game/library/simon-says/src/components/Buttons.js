import styled from "styled-components";
import { Link } from "react-router-dom";

const styles = `
  font-size: 2em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid white;
  color: white;
  background-color: transparent;
  font-family: 'Bungee', sans-serif;
  border-radius: 3px;
  cursor: pointer;
  transition: 0.3s;
  outline: none;
  text-decoration: none;

  &:hover {
    transform: scale(1.02);
  }
`;

export const Button = styled.button`
  ${styles}
`;
export const ButtonLink = styled(Link)`
  ${styles}
`;
