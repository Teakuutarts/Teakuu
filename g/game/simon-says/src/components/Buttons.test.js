import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Button, ButtonLink } from './Buttons';

it('renders Button correctly', () => {
  const wrapper = shallow(
    <Button>foo bar</Button>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders ButtonLink correctly', () => {
  const wrapper = shallow(
    <ButtonLink>foo bar</ButtonLink>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

