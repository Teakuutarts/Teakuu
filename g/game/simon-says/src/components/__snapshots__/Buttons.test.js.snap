exports[`test renders Button correctly 1`] = `
<button
  className="bzjSVi">
  foo bar
</button>
`;

exports[`test renders ButtonLink correctly 1`] = `
<Link
  className="bzjSVi"
  onlyActiveOnIndex={false}
  style={Object {}}>
  foo bar
</Link>
`;
